### PR TITLE
[grid] keep HttpClient alive until unused #12558 

### PR DESCRIPTION
**Pull request by joerg1985**
### Description
The old implementation did close the HttpClient some minutes after access and did not check if the HttpClient has request still waiting for a response. 

### Motivation and Context
This did break long running requests, especially when the client has a high read timeout set. 

**Note:** This PR might make #12975 even worse, as it will not close the http client and prevent the retry to send the command more often.  

**Note:** This PR will also fix the broken get session url endpoint and uses it where ever it makes sense.
This might lead to issues when people are using a grid of mixed versions, but is this supported at all?

~~**Note:** This PR will also add more output when the JSON parser failed, as it was useful to debug the broken get session url endpoint and might be useful for other issues too.~~ has been removed from the PR as requested.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by joerg1985** [comment]
An alternative way to handle this might be to use a WeakReference in the cache and use a Cleaner to call `.close` there.
This could also be used in other areas like the `ProxyWebsocketsIntoGrid` where it is not easy to call `.close` after use.

File: java/src/org/openqa/selenium/json/Input.java [hunk 1]
```diff
@@ -97,7 +97,18 @@ public String toString() {
       length = MEMORY_SIZE;
     }
 
-    return "Last " + length + " characters read: " + new String(buffer, offset, length);
+    String last = "Last " + length + " characters read: " + new String(buffer, offset, length);
```
Comment (pujagani) [inline java/src/org/openqa/selenium/json/Input.java]: This fix can be probably a separate PR.

Comment (joerg1985) [reply java/src/org/openqa/selenium/json/Input.java]: Okay, i will remove this from the PR.
It was helpfull to debug the broken session url endpoint.

**Review by diemol — APPROVED** [review]
Can you please solve the conflict so we merge this?

**Comment by joerg1985** [comment]
@diemol i have fixed it, the PR should have no conflicts left.

**Comment by diemol** [comment]
There are some failures in CI, do you know if they are related?

**Comment by titusfortner** [comment]
latest on trunk broke the CI so that might be causing these failures.

**Comment by joerg1985** [comment]
i am currently looking into it
